pom.xml

```diff
@@ -162,7 +162,9 @@
     <commons.math3.version>3.4.1</commons.math3.version>
     <!-- managed up from 3.2.1 for SPARK-11652 -->
     <commons.collections.version>3.2.2</commons.collections.version>
-    <scala.version>2.12.10</scala.version>
+    <scala-2.12.version>2.12.10</scala-2.12.version>
+    <scala-2.13.version>2.13.5</scala-2.13.version>
```

Comment:
For reviewers:
I confirmed this solution works even for `branch-3.1` but `branch-3.1` uses `2.13.4` rather than `2.13.5` so I'll open a backport PR after this PR merged.

Reply:
Hm, why do we need both versions?

Reply:
I think it's better to have both versions to avoid issues like SPARK-34774.

Reply:
I don't quite see why it is necessary for this change?

Reply:
It's to preserve one version in pom.xml even after overwrite it with`change-scala-version.sh`.
But now that #31865 introduces `<scala.version>` in `<profile>`, we can take another solution without having both `scala-2.12.version>` and `<scala-2.13.version>`.

Reply:
I don't see the advantage or need - surely the build has one scala version? otherwise you have to flip even which property is referenced in the whole build?

Reply:
The first time I open this PR, `<scala.version>2.12.10</scala.version>` is absent from the profile `<id>scala-2.12</id>` so, if we change the version with `change-scala-version.sh 2.13`, there will be no more `2.12.10` because `<scala.version>2.12.10</scala.version>` is overwritten with `<scala.version>2.13.5</scala.version>.
Then, if we try to `change-scala-version.sh 2.12` will fail to modify pom.xml.

If we have both versions in properties (`<scala-2.12.version>` and `<scala-2.13-version>`) which will not be overwritten by `change-scala-version.sh`, it's easy to choose `scala.version` to change.

But, as I mentioned above, now we don't need such properties.